### PR TITLE
depends: remove `FORCE_USE_SYSTEM_CLANG`

### PR DESCRIPTION
**Pull request by fanquake**
Remove `FORCE_USE_SYSTEM_CLANG` in favour of always using the system Clang and lld for macOS cross-compilation; rather than downloading precompiled blobs.

For example, anyone using Ubuntu 24.04 should be able to `apt install clang llvm lld .. etc`, and then cross-compile for macOS using:
```bash
# clang --version
Ubuntu clang version 18.1.3 (1)
make -C depends HOST=arm64-apple-darwin FORCE_USE_SYSTEM_CLANG=1
./autogen.sh
CONFIG_SITE=/path/to/depends/arm64-apple-darwin/share/config.site ./configure
make
# file src/qt/bitcoin-qt
src/qt/bitcoin-qt: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|PIE|HAS_TLV_DESCRIPTORS>
```

Note that the minimum supported version of Clang we will support for macOS cross-compilation will likely be more recent than our other minimum supported version of Clang, due to compiler/linker option usage.

### CODE REVIEW

**Review by theuni — COMMENTED** [review]
Concept ACK

File: depends/README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ The paths are automatically configured and no other options are needed.
 
 #### For macOS cross compilation
 
-    sudo apt-get install curl bsdmainutils cmake zip
+    sudo apt-get install clang cmake curl lld llvm make zip
```
[the remark this answers is not in the document]
Comment (fanquake) [reply depends/README.md:48]: Yea. Depending on when this lands, we could update this to recommend LLVM 18, as that's likely what is going to be used in the CI, if we just `apt install clang` on Ubuntu `24.04`. At a minimum, given we use `-fixup_chains`, we require at least LLVM 16 or later.

[the remark this answers is not in the document]
Comment (fanquake) [reply depends/README.md:48]: Yea. If/when it does change, everything will continue to change in lockstep.

**Comment by hebasto** [comment]
Concept ACK.

**Comment by fanquake** [comment]
Added a commit to rename some references to macho ld64 to lld.

**Comment by Sjors** [comment]
tACK 7cbfd7a7ce45ac68d6041f42f468862f5c193d8c

I made a Guix build, as well as a regular cross-compile on Ubuntu 24.04 for Intel macOS, and lightly tested them.

Guix hashes for darwin:

```
048fa3f381098477b8bea25701547f37a6680b33afb9c61ba88cb4a895aae81d  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/SHA256SUMS.part
d540c4c73b98bef810c85b0b10a6e1a004655fb0c8a4709b8f655c6924afef8b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.tar.gz
4a0bca38e194b731a105ab791d8dfa6adc314fb0a5a963d49136d5546794919b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.zip
993c355cdef9a246c9929dc5b9a734ae4781f6353a8c953465c81ae2739809cd  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin.tar.gz
2e58229e7f9f22448fda71b0eda9fb4f084edb1a44093e0dadf9a2ab0ab37026  guix-build-7cbfd7a7ce45/output/dist-archive/bitcoin-7cbfd7a7ce45.tar.gz
1140b74f33745cdd0e703c7740b3adf79b233be9becca3d464b41fc508ebc3f0  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/SHA256SUMS.part
ae7b6911d7c184e9643f8a611db02c5c5028d116937b42935052dea04cc834cf  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.tar.gz
eea2db60805e1a97ffb9cf412380e5df178dd2f7f5e42e1a06feae3185f87a79  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.zip
90bc2c35c76cce5620fcfeb57f108e02f6be6d7d3ac36c88d9a8f991b9637b7b  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin.tar.gz

```

**Comment by fanquake** [comment]
Guix Build (aarch64):
```bash
f8487a885a04a4b3c273b2d1ba3ef5e100026f16b03c08e866dbf4cd468d0802  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/SHA256SUMS.part
b3da7604bc7302213d2864bddbccc54ede6374711377b89b2d17a989d2e9e64d  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/bitcoin-7cbfd7a7ce45-aarch64-linux-gnu-debug.tar.gz
d2340b2c084faf01a0db8ea0c077ba9f8c51ef23b680396d88001b33d126788b  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/bitcoin-7cbfd7a7ce45-aarch64-linux-gnu.tar.gz
a79f460ac8b4c893ad1cf4aba17489b0dc7834377166d4e9bc4da78bcd84a430  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/SHA256SUMS.part
6d94b11fca39489230f00125d156c1ad36a8b8ff9c5fdd6a7d47e31c3c527230  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/bitcoin-7cbfd7a7ce45-arm-linux-gnueabihf-debug.tar.gz
e5cf9b6bd1a8ad95a49ad523681091ad9938c02f7f995934c413bd8b8c651199  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/bitcoin-7cbfd7a7ce45-arm-linux-gnueabihf.tar.gz
048fa3f381098477b8bea25701547f37a6680b33afb9c61ba88cb4a895aae81d  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/SHA256SUMS.part
d540c4c73b98bef810c85b0b10a6e1a004655fb0c8a4709b8f655c6924afef8b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.tar.gz
4a0bca38e194b731a105ab791d8dfa6adc314fb0a5a963d49136d5546794919b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.zip
993c355cdef9a246c9929dc5b9a734ae4781f6353a8c953465c81ae2739809cd  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin.tar.gz
2e58229e7f9f22448fda71b0eda9fb4f084edb1a44093e0dadf9a2ab0ab37026  guix-build-7cbfd7a7ce45/output/dist-archive/bitcoin-7cbfd7a7ce45.tar.gz
18116e5c3df56d933ce453dd9b475317fe3a9e4452fa420e57cfeb15f6dd0580  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/SHA256SUMS.part
d23d0c154fa27210e0be3992bd34882db469dde51070b975cb5ee6b365bfc10b  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/bitcoin-7cbfd7a7ce45-powerpc64-linux-gnu-debug.tar.gz
92d17f3834ee8f8f14cea0ff7e183c8efb1a17355d4fd654ac0a96e01c880aa3  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/bitcoin-7cbfd7a7ce45-powerpc64-linux-gnu.tar.gz
3bd64df4a6e6fd27ac72c48691e41a5b528446a4052c42caaf6e91f32419d327  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/SHA256SUMS.part
dd23855a2f58b3334b274630b7c97be99335ff9908e8a6b0c1300a141b92f5a0  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/bitcoin-7cbfd7a7ce45-riscv64-linux-gnu-debug.tar.gz
a40be60dd17b32b6c8c8560e579bfb02c08e2e8f082a2247b66f957b2780fee9  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/bitcoin-7cbfd7a7ce45-riscv64-linux-gnu.tar.gz
1140b74f33745cdd0e703c7740b3adf79b233be9becca3d464b41fc508ebc3f0  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/SHA256SUMS.part
ae7b6911d7c184e9643f8a611db02c5c5028d116937b42935052dea04cc834cf  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.tar.gz
eea2db60805e1a97ffb9cf412380e5df178dd2f7f5e42e1a06feae3185f87a79  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.zip
90bc2c35c76cce5620fcfeb57f108e02f6be6d7d3ac36c88d9a8f991b9637b7b  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin.tar.gz
8b758e8b739b80123a132ceba172ba9fd70a476941a1b515629737b9f6020ab9  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/SHA256SUMS.part
cd50dfb8d19c5ea62b1069655c5154965703fccfa31bad9c7169b2842a5c5048  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/bitcoin-7cbfd7a7ce45-x86_64-linux-gnu-debug.tar.gz
1496518bd7efe42df95318666a14c97ae019a6d66c526906c61d22dd8769769d  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/bitcoin-7cbfd7a7ce45-x86_64-linux-gnu.tar.gz
56d85879d391893a174dce042ed7a4d54c4e57645629660952411052e35d793c  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/SHA256SUMS.part
633ae51f73e818edcfd7e6a03d6368abe58540a485908aa17dbd812cc790d55f  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-debug.zip
480576fb02cc43bcaea583b4975e7609c093c53a16c891e1ab1c75801cd9da18  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-setup-unsigned.exe
ff2e48fb53aca49d3d0fcda480e64cf36f4b8250357c1adeec0abba6b66bb524  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-unsigned.tar.gz
42ca2d89cb6703094ea22bd9a869d833f36f42cbef39ca5c3c7b8503bbe15587  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64.zip
```

**Review by theuni — APPROVED** [review]
Tested on Ubuntu 22.04 with downloaded llvm 18 binaries in PATH.

ACK 7cbfd7a7ce45ac68d6041f42f468862f5c193d8c

**Review by sedited — APPROVED** [review]
Nice, ACK 7cbfd7a7ce45ac68d6041f42f468862f5c193d8c


Guix builds (aarch64):
```
f8487a885a04a4b3c273b2d1ba3ef5e100026f16b03c08e866dbf4cd468d0802  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/SHA256SUMS.part
b3da7604bc7302213d2864bddbccc54ede6374711377b89b2d17a989d2e9e64d  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/bitcoin-7cbfd7a7ce45-aarch64-linux-gnu-debug.tar.gz
d2340b2c084faf01a0db8ea0c077ba9f8c51ef23b680396d88001b33d126788b  guix-build-7cbfd7a7ce45/output/aarch64-linux-gnu/bitcoin-7cbfd7a7ce45-aarch64-linux-gnu.tar.gz
a79f460ac8b4c893ad1cf4aba17489b0dc7834377166d4e9bc4da78bcd84a430  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/SHA256SUMS.part
6d94b11fca39489230f00125d156c1ad36a8b8ff9c5fdd6a7d47e31c3c527230  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/bitcoin-7cbfd7a7ce45-arm-linux-gnueabihf-debug.tar.gz
e5cf9b6bd1a8ad95a49ad523681091ad9938c02f7f995934c413bd8b8c651199  guix-build-7cbfd7a7ce45/output/arm-linux-gnueabihf/bitcoin-7cbfd7a7ce45-arm-linux-gnueabihf.tar.gz
048fa3f381098477b8bea25701547f37a6680b33afb9c61ba88cb4a895aae81d  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/SHA256SUMS.part
d540c4c73b98bef810c85b0b10a6e1a004655fb0c8a4709b8f655c6924afef8b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.tar.gz
4a0bca38e194b731a105ab791d8dfa6adc314fb0a5a963d49136d5546794919b  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin-unsigned.zip
993c355cdef9a246c9929dc5b9a734ae4781f6353a8c953465c81ae2739809cd  guix-build-7cbfd7a7ce45/output/arm64-apple-darwin/bitcoin-7cbfd7a7ce45-arm64-apple-darwin.tar.gz
2e58229e7f9f22448fda71b0eda9fb4f084edb1a44093e0dadf9a2ab0ab37026  guix-build-7cbfd7a7ce45/output/dist-archive/bitcoin-7cbfd7a7ce45.tar.gz
18116e5c3df56d933ce453dd9b475317fe3a9e4452fa420e57cfeb15f6dd0580  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/SHA256SUMS.part
d23d0c154fa27210e0be3992bd34882db469dde51070b975cb5ee6b365bfc10b  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/bitcoin-7cbfd7a7ce45-powerpc64-linux-gnu-debug.tar.gz
92d17f3834ee8f8f14cea0ff7e183c8efb1a17355d4fd654ac0a96e01c880aa3  guix-build-7cbfd7a7ce45/output/powerpc64-linux-gnu/bitcoin-7cbfd7a7ce45-powerpc64-linux-gnu.tar.gz
3bd64df4a6e6fd27ac72c48691e41a5b528446a4052c42caaf6e91f32419d327  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/SHA256SUMS.part
dd23855a2f58b3334b274630b7c97be99335ff9908e8a6b0c1300a141b92f5a0  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/bitcoin-7cbfd7a7ce45-riscv64-linux-gnu-debug.tar.gz
a40be60dd17b32b6c8c8560e579bfb02c08e2e8f082a2247b66f957b2780fee9  guix-build-7cbfd7a7ce45/output/riscv64-linux-gnu/bitcoin-7cbfd7a7ce45-riscv64-linux-gnu.tar.gz
1140b74f33745cdd0e703c7740b3adf79b233be9becca3d464b41fc508ebc3f0  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/SHA256SUMS.part
ae7b6911d7c184e9643f8a611db02c5c5028d116937b42935052dea04cc834cf  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.tar.gz
eea2db60805e1a97ffb9cf412380e5df178dd2f7f5e42e1a06feae3185f87a79  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin-unsigned.zip
90bc2c35c76cce5620fcfeb57f108e02f6be6d7d3ac36c88d9a8f991b9637b7b  guix-build-7cbfd7a7ce45/output/x86_64-apple-darwin/bitcoin-7cbfd7a7ce45-x86_64-apple-darwin.tar.gz
8b758e8b739b80123a132ceba172ba9fd70a476941a1b515629737b9f6020ab9  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/SHA256SUMS.part
cd50dfb8d19c5ea62b1069655c5154965703fccfa31bad9c7169b2842a5c5048  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/bitcoin-7cbfd7a7ce45-x86_64-linux-gnu-debug.tar.gz
1496518bd7efe42df95318666a14c97ae019a6d66c526906c61d22dd8769769d  guix-build-7cbfd7a7ce45/output/x86_64-linux-gnu/bitcoin-7cbfd7a7ce45-x86_64-linux-gnu.tar.gz
56d85879d391893a174dce042ed7a4d54c4e57645629660952411052e35d793c  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/SHA256SUMS.part
633ae51f73e818edcfd7e6a03d6368abe58540a485908aa17dbd812cc790d55f  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-debug.zip
480576fb02cc43bcaea583b4975e7609c093c53a16c891e1ab1c75801cd9da18  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-setup-unsigned.exe
ff2e48fb53aca49d3d0fcda480e64cf36f4b8250357c1adeec0abba6b66bb524  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64-unsigned.tar.gz
42ca2d89cb6703094ea22bd9a869d833f36f42cbef39ca5c3c7b8503bbe15587  guix-build-7cbfd7a7ce45/output/x86_64-w64-mingw32/bitcoin-7cbfd7a7ce45-win64.zip
```

**Comment by hebasto** [comment]
CI related stuff has to be ported to the CMake staging branch.

**Comment by hebasto** [comment]
Ported to the CMake-based build system in https://github.com/hebasto/bitcoin/pull/264.

**Pull request merged**